### PR TITLE
Feature improve release docs part2

### PR DESCRIPTION
**Pull request by johnatasr**
#### Proposed Changes ####

- Changes for K3s release patch docs

#### Types of Changes ####

- Documentation

#### Testing ####

Fixed docker volume permission errors:

dockervolumefix

dockervolumefix2

#### Linked Issues ####

https://github.com/rancher/ecm-distro-tools/issues/210
https://github.com/rancher/ecm-distro-tools/issues/211
https://github.com/rancher/ecm-distro-tools/issues/212

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: docs/release/kubernetes-upgrade.md [hunk 1]
```diff
@@ -4,6 +4,37 @@ This document details the K3S kubernetes patch release process.
...
+- Create or edit the global git configuration file by running the following command:
+   
+   ```shell
+   git config --global --edit
+  ```
+- In the configuration file, add the following line under the [core] section:
+  ```sh 
+  [core]
+    safelyUseIncompatibleGitCredentialHelper = true
+  ```
```
Comment (Tashima42) [inline docs/release/kubernetes-upgrade.md:32]: It's easier to just run the command:
```bash
git config --global core.safelyUseIncompatibleGitCredentialHelper true
```

File: docs/release/kubernetes-upgrade.md [hunk 2]
```diff
@@ -4,6 +4,37 @@ This document details the K3S kubernetes patch release process.
...
+
+This setting ensures that the container user can safely access the Git repository directory without permission issues.
+
+Note: If you are using a specific git configuration file for the project, make sure to add the `safelyUseIncompatibleGitCredentialHelper = true` line in that file instead.
```
Comment (Tashima42) [inline docs/release/kubernetes-upgrade.md:36]: Unless the user specified otherwise, it should be ok to not do this.

> Each of these “levels” (system, global, local) overwrites values in the previous level, so values in .git/config trump those in [path]/etc/gitconfig, for instance.
https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration

```suggestion
```

File: docs/release/kubernetes-upgrade.md [hunk 3]
```diff
@@ -4,6 +4,30 @@ This document details the K3S kubernetes patch release process.
...
+
+#####  User Mismatch in Docker Container
+
+When running the script inside a Docker container, there might be a user mismatch between the container user and the git repository directory user. This can result in permission issues when performing Git operations.
+
+To avoid this problem, you can add a safe directory setting in the git configuration by following these steps:
+
+- Create or edit the global git configuration file by running the following command:
+   
+   ```shell
```
Comment (briandowns) [inline docs/release/kubernetes-upgrade.md:25]: Can you remove the whitespace preceding these lines?

Comment (briandowns) [reply docs/release/kubernetes-upgrade.md:25]: Same as above as well.

File: docs/release/kubernetes-upgrade.md [hunk 4]
```diff
@@ -71,9 +102,9 @@ echo -e ${BUILD_CONTAINER} | docker build -t ${GOIMAGE}-dev -
 # Now create the tags by executing tag.sh with the given version variables.
 docker run --rm -u $(id -u) \
 --mount type=tmpfs,destination=${GOPATH}/pkg \
--v ${GOPATH}/src:/go/src \
--v ${GOPATH}/.cache:/go/.cache \
--v ${GLOBAL_GIT_CONFIG_PATH}:/go/.gitconfig \
+-v ${GOPATH}/src:/go/src:rw \
```
[the remark this answers is not in the document]
Comment (briandowns) [reply docs/release/kubernetes-upgrade.md:97]: What's the end result here? Do we want to include any of these options in the PR?

File: docs/release/kubernetes-upgrade.md [hunk 5]
```diff
@@ -182,14 +206,16 @@ Once CI passes and you receive two approvals, you may now squash-merge the PR an
...
-1. Set title and tag according to the release version you're working on. E.g. v1.22.5-rc1+k3s1.
+1. Set title and tag according to the release version you're working on. E.g. v1.27.4-rc1+k3s1.
 2. Leave description blank.
 3. Check the pre-release field.
 4. Publish
 
 The resulting run can be viewed here: 
 [k3s-io/k3s Drone Dashboard](https://drone-publish.k3s.io/k3s-io/k3s)
 
+It may be necessary to open a new release candidate to accommodate changes in dependencies, such as modifications in https://github.com/k3s-io/k3s-upgrade. To do this, simply repeat the previous process but open it as, for example, `v1.27.4-rc2+k3s1`.
```
Comment (briandowns) [inline docs/release/kubernetes-upgrade.md:217]:
```suggestion
It may be necessary to create a new release candidate to accommodate changes in dependencies such as modifications in https://github.com/k3s-io/k3s-upgrade. To do this, repeat the previous process for tagging and increment the rc version.
```

File: docs/release/kubernetes-upgrade.md [hunk 6]
```diff
@@ -85,36 +108,36 @@ chmod +x push.sh
...
-git push ${REMOTE} staging/src/k8s.io/kubectl/v1.27.3-k3s1
-git push ${REMOTE} staging/src/k8s.io/kubelet/v1.27.3-k3s1
-git push ${REMOTE} staging/src/k8s.io/legacy-cloud-providers/v1.27.3-k3s1
-git push ${REMOTE} staging/src/k8s.io/metrics/v1.27.3-k3s1
-git push ${REMOTE} staging/src/k8s.io/mount-utils/v1.27.0-k3s1
-git push ${REMOTE} staging/src/k8s.io/sample-apiserver/v1.27.3-k3s1
-git push ${REMOTE} staging/src/k8s.io/sample-cli-plugin/v1.27-k3s1
-git push ${REMOTE} staging/src/k8s.io/sample-controller/v1.27.3-k3s1
-git push ${REMOTE} v1.27.3-k3s1
+git push $REMOTE staging/src/k8s.io/api/v1.27.4-k3s1
```
Comment (briandowns) [inline docs/release/kubernetes-upgrade.md:111]: Please add the brackets back.

File: docs/release/kubernetes-upgrade.md [hunk 7]
```diff
@@ -259,10 +293,7 @@ A later version can point to those arguments with no change:
     serverArgs: *serverArgs-v1
 ```
 If you are unsure of the new minor versions min/max constraints you can ask the Project manager and/or QA.
-# Create system-agent-installer-k3s Release Images
```
Comment (briandowns) [inline docs/release/kubernetes-upgrade.md:262]: I think this is useful but maybe rephrased since it's not a manual step but something to check on and verify the tag was created.
